File: LeetCode/Blind 75/TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public int[] twoSumUsingMap(int[] nums, int target) {
        Map<Integer, Integer> valueIdxMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (valueIdxMap.containsKey(diff))
                return new int[] { valueIdxMap.get(diff), i };
            valueIdxMap.put(nums[i], i);
        }
        return new int[] { -1, -1 };
    }
}

File: LeetCode/Blind 75/TwoSum.py
class TwoSum():
    def two_sum(self, nums, target):
        value_idx_map = dict()
        for i, num in enumerate(nums):
            diff = target - num
            if diff in value_idx_map:
                return [value_idx_map[diff], i]
            value_idx_map[num] = i

        return [-1, -1]
